### DISCUSSION TITLE
How can I set Transition duration as resource?

### DISCUSSION THREAD

**Discussion by XDgl233** [AvaloniaUI/Avalonia#12233] [Q&A] [answered]
I want to control my generic controls' transition duration time in a resource dictionary, but I don't know what to do. I can't find something like Duration or do I need to store the entire Transitions /Transitions blocks in the resource file? And if I just want to store some base type data (like double, int, etc), what can I do with the resource file or should I avoid doing these?
I am a new learner, and would really appreciate your advice and help :)

**Comment by danielmayost** [comment, marked as answer]
yes, it is accepted and it is also done in the source files in Avalonia.

```
<ResourceDictionary xmlns="https://github.com/avaloniaui"
                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
                    xmlns:sys="using:System">
  <x:Boolean x:Key="Var1">True</x:Boolean>
  <sys:Double x:Key="Var2">5</sys:Double>
  <sys:TimeSpan x:Key="Var3">00:00:00.1</sys:TimeSpan>
</ResourceDictionary>
```

```
Duration="{DynamicResource Var3}"
```

**Reply by XDgl233** [reply]
Thank you so much! :)

Edit: I tried using the timespan in your example, but it still gives me the error:

Unable to find suitable setter or adder for property Duration of type Avalonia.Base:Avalonia.Animation.Transition`1 for argument Avalonia.Base:Avalonia.Data.IBinding, available setter parameter lists are:
System.TimeSpan
System.TimeSpan Line 43, position 41.

But other resource like using boolean and double work smoothly :)

**Reply by danielmayost** [reply]
> but it still gives me the error

you right. the `Duration` property isn't `AvaloniaProperty` so you can use just only StaticResource (which will not respond to changes):
```
Duration="{StaticResource Var3}"
```
